packages/auto-approve/test/check-config.test.ts

```diff
@@ -0,0 +1,300 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  validateYaml,
+  validateSchema,
+  checkCodeOwners,
+} from '../src/check-config.js';
+import {describe, it} from 'mocha';
+import assert from 'assert';
+import * as fs from 'fs';
+import yaml from 'js-yaml';
+import nock from 'nock';
+const {Octokit} = require('@octokit/rest');
+
+const octokit = new Octokit({
+  auth: 'mypersonalaccesstoken123',
+});
+const CONFIGURATION_FILE_PATH = 'auto-approve.yml';
+
+nock.disableNetConnect();
+
+function getCodeOwnersFile(response: string | undefined, status: number) {
+  return nock('https://api.github.com')
+    .get('/repos/owner/repo/contents/.github%2FCODEOWNERS')
+    .reply(
+      status,
+      response ? {content: Buffer.from(response).toString('base64')} : undefined
+    );
+}
+
+async function invalidateSchema(configNum: number) {
+  return await validateSchema(
+    yaml.load(
+      fs.readFileSync(
+        `./test/fixtures/config/invalid-schemas/invalid-schema${configNum}.yml`,
+        'utf8'
+      )
+    )
+  );
+}
+
+describe('check for config', () => {
+  describe('whether config is a valid YAML object', () => {
+    it('should return error message if YAML is invalid', () => {
+      const isYamlValid = validateYaml(
+        fs.readFileSync(
+          './test/fixtures/config/invalid-schemas/invalid-yaml-config.yml',
+          'utf8'
+        )
+      );
+      assert.strictEqual(
+        isYamlValid.message,
+        'File is not properly configured YAML'
+      );
+    });
+
+    it('should return true if YAML is valid', async () => {
+      const isYamlValid = validateYaml(
+        fs.readFileSync(
+          './test/fixtures/config/valid-schemas/valid-schema1.yml',
+          'utf8'
+        )
+      );
+      assert.strictEqual(isYamlValid.isValid, true);
+    });
+  });
+
+  describe('whether YAML file has valid schema', async () => {
+    it('should fail if YAML has any other properties than the ones specified', async () => {
+      //does not have any additional properties
+      assert.deepStrictEqual((await invalidateSchema(1)).errorMessages, [
+        {
+          wrongProperty: {additionalProperty: 'notTheRules'},
+          message: 'should NOT have additional properties',
+        },
+      ]);
+    });
+
+    it('should fail if title does not match first author', async () => {
+      //title does not correspond to author
+      assert.deepStrictEqual((await invalidateSchema(2)).errorMessages, [
+        {
+          wrongProperty: {allowedValue: '^chore: regenerate README$'},
+          message: 'should be equal to constant',
+        },
+      ]);
+    });
+
+    it('should fail if title does not match second author', async () => {
+      //title does not correspond to author
+      assert.deepStrictEqual((await invalidateSchema(3)).errorMessages, [
```

Comment:
Asserting that the error text exactly match given error text makes for a very fragile test.  If I need to fix a typo in the error message later, I then need to fix tests too.

There's always a trade off between writing tests that confirm every detail and writing tests that are easy to maintain.

A less fragile test confirms the function returns *any* error message for the given test.  That's still going to confirm bad input is reported, but be smaller burden to maintain.

Reply:
I thought a bit about this suggestion, and I implemented what I think might be halfway point: https://www.npmjs.com/package/npm-snapshot

I'm using this library to confirm that the error message is the same as it was last time. It's less work to change in the future (you just have to delete `snapshots` and then rerun tests), but it can provide some more specific checking.